### PR TITLE
Done

### PR DESCRIPTION
**Pull request by vovkandrew**

### CODE REVIEW

File: src/main/java/core/basesyntax/Rectangle.java [hunk 1]
```diff
@@ -27,4 +27,30 @@ public Integer getLength() {
...
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        int generatedHashCode = 9;
+        generatedHashCode = generatedHashCode * 19 + width;
+        generatedHashCode = generatedHashCode * 19 + length;
+        generatedHashCode = generatedHashCode * 19 + color.hashCode();
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Rectangle.java]: what happens, if color == null ?

File: src/main/java/core/basesyntax/Rectangle.java [hunk 2]
```diff
@@ -27,4 +27,30 @@ public Integer getLength() {
...
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+        if (object == null) {
+            return false;
+        }
+        if (object.getClass().equals(Rectangle.class)) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Rectangle.java]:
```suggestion
        if (object.getClass().equals(getClass())) {
```

File: src/main/java/core/basesyntax/Rectangle.java (hunk 1 above)
Comment (vovkandrew) [reply src/main/java/core/basesyntax/Rectangle.java]: then we will add just 0 as hashCode(null) = 0, is that prohibited?

Comment (RomanDubovskyi) [reply src/main/java/core/basesyntax/Rectangle.java]: you sure that there won't be nullPointerException if color == null? 
it's not hashCode(null) it's null.hashCode()

Comment (vovkandrew) [reply src/main/java/core/basesyntax/Rectangle.java]: null check added, I think that may help

File: src/main/java/core/basesyntax/Rectangle.java [hunk 3]
```diff
@@ -27,4 +27,33 @@ public Integer getLength() {
...
+
+    @Override
+    public int hashCode() {
+        int generatedHashCode = 9;
+        generatedHashCode = generatedHashCode * 19 + width;
+        generatedHashCode = generatedHashCode * 19 + length;
+        if (color == null) {
+            generatedHashCode = generatedHashCode * 19;
+        }
+        generatedHashCode = generatedHashCode * 19 + color.hashCode();
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Rectangle.java]: did you test your current solution with color = null? 
cause line after if statement is being executed anyway

Comment (vovkandrew) [reply src/main/java/core/basesyntax/Rectangle.java]: Oh, missed that, have already fixed!

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request closed without merging**